pkg/api/server/v1alpha2/log/s3.go

```diff
@@ -0,0 +1,225 @@
+package log
+
+import (
+	"bufio"
+	"bytes"
+	"path/filepath"
+
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	server "github.com/tektoncd/results/pkg/api/server/config"
+	"github.com/tektoncd/results/pkg/apis/v1alpha2"
+)
+
+const (
+	DefaultS3MultiPartSize = 1024 * 1024 * 5
+)
+
+type s3Client interface {
+	AbortMultipartUpload(context.Context, *s3.AbortMultipartUploadInput, ...func(*s3.Options)) (*s3.AbortMultipartUploadOutput, error)
+	CompleteMultipartUpload(context.Context, *s3.CompleteMultipartUploadInput, ...func(*s3.Options)) (*s3.CompleteMultipartUploadOutput, error)
+	CreateMultipartUpload(context.Context, *s3.CreateMultipartUploadInput, ...func(*s3.Options)) (*s3.CreateMultipartUploadOutput, error)
+	DeleteObject(context.Context, *s3.DeleteObjectInput, ...func(*s3.Options)) (*s3.DeleteObjectOutput, error)
+	GetObject(context.Context, *s3.GetObjectInput, ...func(*s3.Options)) (*s3.GetObjectOutput, error)
+	UploadPart(context.Context, *s3.UploadPartInput, ...func(*s3.Options)) (*s3.UploadPartOutput, error)
```

Comment:
This is the right pattern - the new AWS SDK recommends using your own interfaces to mock/stub in unit tests:

https://aws.github.io/aws-sdk-go-v2/docs/unit-testing/

Reply:
Added unit tests covering basic cases for S3